docs/tutorials/streaming_analytics.md

```diff
@@ -0,0 +1,460 @@
+yet arrive. In fact, you have the benefit of some god-like knowledge of this stream's future, and
+you can see that your stream sorter should wait at least until the 2 arrives before producing any
+results.
+
+*Some buffering, and some delay, is necessary.*
+
+(2) If you do this wrong, you could end up waiting forever. First the sorter saw an event from time
+4, and then an event from time 2. Will an event with a timestamp less than 2 ever arrive? Maybe.
+Maybe not. You could wait forever and never see a 1.
+
+*Eventually you have to be courageous and emit the 2 as the start of the sorted stream.*
+
+(3) What you need then is some sort of policy that defines when, for any given timestamped event, to
+stop waiting for the arrival of earlier events.
+
+*This is precisely what watermarks do* — they define when to stop waiting for earlier events.
+
+Event time processing in Flink depends on *watermark generators* that insert special timestamped
+elements into the stream, called *watermarks*. A watermark for time _t_ is an assertion that the
+stream is (probably) now complete up through time _t_.
+
+When should this stream sorter stop waiting, and push out the 2 to start the sorted stream? When a
+watermark arrives with a timestamp of 2, or greater.
+
+(4) You might imagine different policies for deciding how to generate watermarks.
+
+Each event arrives after some delay, and these delays vary, so some events are delayed more than
+others. One simple approach is to assume that these delays are bounded by some maximum delay. Flink
+refers to this strategy as *bounded-out-of-orderness* watermarking. It's easy to imagine more
+complex approaches to watermarking, but for most applications a fixed delay works well enough.
+
+### Latency vs. Completeness
+
+Another way to think about watermarks is that they give you, the developer of a streaming
+application, control over the tradeoff between latency and completeness. Unlike in batch processing,
+where one has the luxury of being able to have complete knowledge of the input before producing any
+results, with streaming you must eventually stop waiting to see more of the input, and produce some
+sort of result.
+
+You can either configure your watermarking aggressively, with a short bounded delay, and thereby
+take the risk of producing results with rather incomplete knowledge of the input -- i.e., a possibly
+wrong result, produced quickly. Or you can wait longer, and produce results that take advantage of
+having more complete knowledge of the input stream(s).
+
+It is also possible to implement hybrid solutions that produce initial results quickly, and then
+supply updates to those results as additional (late) data is processed. This is a good approach for
+some applications.
+
+### Lateness
+
+Lateness is defined relative to the watermarks. A `Watermark(t)` asserts that the stream is complete
+up through time _t_; any event following this watermark whose timestamp is &le; _t_ is late.
```

Comment:
I think, it should be this, shouldn't it?
```suggestion
up thru time _t_; any event following this watermark whose timestamp is &le; _t_ is late.
```

Reply:
"Through is the only formally accepted spelling of the word. Thru is an alternate spelling that should be used only in informal writing or when referring to drive-throughs."

Reply:
oh, I guess, then the other "thru" that I saw in a previous PR may need changing...(I was wondering there, looked it up quickly, and then proposed it here as a result)